tests/integration/gopackagesdriver/gopackagesdriver_test.go

```diff
@@ -124,3 +147,37 @@ func TestBaseFileLookup(t *testing.T) {
 		}
 	})
 }
+
+func TestExternalTests(t *testing.T) {
+	reader := strings.NewReader("{}")
+	out, _, err := bazel_testing.BazelOutputWithInput(reader, "run", "@io_bazel_rules_go//go/tools/gopackagesdriver", "--", "file=hello_external_test.go")
+	if err != nil {
+		t.Errorf("Unexpected error: %w", err.Error())
+	}
+	var resp response
+	err = json.Unmarshal(out, &resp)
+	if err != nil {
+		t.Errorf("Failed to unmarshal packages driver response: %w\n%w", err.Error(), out)
+	}
+
+	if len(resp.Roots) != 2 {
+		t.Errorf("Expected exactly two roots for package: %+v", resp.Roots)
+	}
+
+	var testId, xTestId string
+	for _, id := range resp.Roots {
+		if strings.HasSuffix(id, "_xtest") {
+			xTestId = id
+		} else {
+			testId = id
+		}
+	}
+
+	for _, p := range resp.Packages {
+		if p.ID == xTestId && !strings.HasSuffix(p.GoFiles[0], "external_test.go") {
+			t.Errorf("Expected only one file in _xtest pkg, got %+v", p.GoFiles)
```

Comment:
It looks like we aren't testing the cardinality here.

Reply:
We are only supposed to have one file in here, not sure about the necessity to check cardinality. Will add a len() == 1 in the check as well.